source/slang/slang.cpp

```diff
@@ -170,6 +170,7 @@ void Session::init()
     // Built in linkage uses the built in builder
     m_builtinLinkage = new Linkage(this, builtinAstBuilder, nullptr);
     m_builtinLinkage->m_optionSet.set(CompilerOptionName::DebugInformation, DebugInfoLevel::None);
+    m_builtinLinkage->m_optionSet.set(CompilerOptionName::DisableShortCircuit, true);
```

Comment:
Let's keep the default to not disable.

Reply:
I noticed that if we don't disable the short-circuit in stdlib, then the user code could be overload by the builtin  "&&".
So something like
`bool = int && int`
will still be short-circuiting.
How can we not allow such overload?

Reply:
Why choosing && would lead to short circuiting?

Reply:
Because the stdlib defines "&&" operator which is compiled by enabling short-circuit.

Reply:
Won't this cause issue? Is `and(T1, T2)` defined somewhere else?

Reply:
https://github.com/shader-slang/slang/pull/4056
New PR is trying to fix this issue.